### DISCUSSION TITLE
HTTP 500 while adding new assignment and potential grader bug

### DISCUSSION THREAD

**Discussion by rlalik** [community/community#125198] [GitHub Education]
### Hi everyone,

Hi,

while adding an assignment in the autograder section, I spotted some weird behavior which in the final resulted in HTTP 500 error, which now reappears each time I am trying to add an assignment. So steps I did.

In the thirds step I wanted to add some automatic grading tests.
1. In the dropdow list I choosed python, gave the suggested name "Python test", install command "pip install pytest".
2. Then I added the second test for pylint using `run_command` option from the list. I named in "Linter". At this time I realized I can assign a points to the successful tests.
3. I wanted to return to my "Python Test" and also assign points. I edited the test, updated the points input box, and saved.
4. At this moment I realized that in my tests list I have twice the position "Linter" whereas I expected "Python Test" and "Linter".
5. Clicking the edit button for each was opening the edit for "Linter". I could not see "Python Test" any more.
6. I tried just to go back to previous page in the wizard, and then go again to the grader page. But at this moment I got 500 error. The message says:

> Internal server error
> We're sorry, but something went wrong.
> [Try going back to classroom dashboard](https://classroom.github.com/)

So I go back to classrom, I enter my class, I try to start the assignment again, it already has pre-filled inputs from previous assignment adding, it works for the first and the second step, but again, fails with 500 for the this one.

Seems like something got cached.

**Comment by bharlowRHC** [comment]
I ran into this several times yesterday and today. The only way I could create the new assignment without the Error 500 was to not put any of the Auto Grading tests in the assignment during the creation process. Then after I created it, I could add the tests cases.

This worked fine so long as I did not want to use the **Reuse Assignment** feature. I ran into a different issue with that an Auto Grading.

**Comment by vishal11u** [comment]
I ran into this several times yesterday and today. The only way I could create the new assignment without the Error 500 was to not put any of the Auto Grading tests in the assignment during the creation process. Then after I created it, I could add the tests cases.

This worked fine so long as I did not want to use the Reuse Assignment feature. I ran into a different issue with that an Auto Grading.
